R/importM.COSMIC.R

```diff
@@ -0,0 +1,348 @@
+    ##                 "cosmic", type = "Cosmic",
+    ##                 outputName = "../inst/extdata/fastMapCosmic.rds")
+
+    # load fastMap hash table
+    fastMapCosmic <- readRDS(system.file("extdata",
+                                       "fastMapCosmic.rds",
+                                       package = "rete"))
+
+    # Validate that the type of the hash table is "Cosmic"
+    if (!is.null(attributes(fastMapCosmic)$type)) {
+        if ("Cosmic" != attributes(fastMapCosmic)$type) {
+            stop(sprintf("Expected hash table type: Cosmic.\nSupplied hash table type: %s",
+                         attributes(fastMapCosmic)$type))
+        }
+    } else {
+        stop("Supplied hash table does not have a type attribute.")
+    }
+
+
+    # ==== CONSTRUCT rCNA OBJECT ==============================================
+
+    if (!silent) {
+        cat("Generating rCNA object using CNA data in ", fNameCNV, ".\n")
+    }
+
+    barcodes <- c()
+    hgncSym <- c()
+
+    for (i in 1:nrow(dataCNV)) {
+
+        # Generate all barcodes for the mutations in dataCNV
+        barcodes <- append(barcodes,
+                           paste("TCGA-",
+                                 dataCNV[i,]$ID_SAMPLE, "-",
+                                 dataCNV[i,]$ID_TUMOUR,
+                                 sep = ""))
+
+        # Generate all HGNC symbols using fastMap()
+        hgncSym <- append(hgncSym,
+                          fastMap(dataCNV[i,]$ID_GENE,
+                                  fastMapCosmic,
+                                  type = "Cosmic",
+                                  dev = TRUE))
+
+    }
+
+    # Create a matrix for storing CNV counts.
+    #     nrow is equal to number of unique HGNC symbols
+    #     ncol is equal to the number of unique barcodes.
+    rCNA <- matrix(nrow = length(unique(hgncSym)),
+                   ncol = length(unique(barcodes)))
+
+    # Set the row and column names of rCNA to the
+    # unique HGNC symbols and unique barcodes
+    rownames(rCNA) <- unique(hgncSym)
+    colnames(rCNA) <- unique(barcodes)
+
+    # Populate rCNA matrix with CNA counts
+    for (i in 1:nrow(dataCNV)) {
+        rCNA[fastMap(dataCNV[i,]$ID_GENE,
+                     fastMapCosmic,
+                     type = "Cosmic",
+                     dev = TRUE),
+             paste("TCGA-",
+                   dataCNV[i,]$ID_SAMPLE, "-",
+                   dataCNV[i,]$ID_TUMOUR,
+                   sep = "")] <- as.numeric(dataCNV[i,]$TOTAL_CN)
+    }
+
+    if (!silent) {
+        cat("rCNA object successfully generated.\n")
+    }
+
+
+    # ==== WRITE TO LOG =======================================================
+
+    if(writeLog) {
+
+        myTitle <- "importM.COSMIC"
+
+        # Compile function call record
+        myCall <- character()
+        myCall[1] <- "importM.COSMIC("
+        myCall[2] <- sprintf("fnameCNV = \"%s\", ", fNameCNV)
+        if (writeToFile) {
```

Comment:
Hm. I know that you've coupled the two arguments in your parameter checks - but I would not enforce the dependency here but just report the parameters as they appear at this point. So even if !writeToFile, I would write the outFname. Reason is, normally you don't even get to this - because you have already checked, but if we ever change the programming logic, these two parts of the code become inconsistent. What you would check at this point is only ```missing()``` for parameters that are allowed to be unset.